### DISCUSSION TITLE
Adding Span to a method invoked through Reflection

### DISCUSSION THREAD

**Discussion by rumpfc** [open-telemetry/opentelemetry-java-instrumentation#9807] [Q&A]
I use a class with annotated methods. Those methods shall be invoked (and traced). Unfortunately the OpenTelemetry javaagent (1.31.0) is not tracing any methods run through `Method.invoke(...)`.

Below is a small sample project (Spring Boot). For demonstration, I just have some methods without any parameters or return types.

```java
class Sample {
    public Sample() {
        // prepare everything
    }

    @WithSpan
    @LogicExecutor
    public void logic1() {
        // do something
    }

    @WithSpan
    @LogicExecutor
    public void logic2() {
        // do something
    }
}

/* === Invoking Logic === */

@Component
class LogicInvoker {

    @WithSpan
    public invokeLogicExeutors() {
        Sample1 sample = new Sample1();
        List<Method> logicExecutors = Arrays.stream(Sample1.class.getDeclaredMethods())
                .filter(m -> m.isAnnotationPresent(LogicExecutor.class))
                .toList();

        for (Method logicExecutorMethod : logicExecutors) {
            try {
                logicExecutorMethod.invoke(sample);
            } catch (InvocationTargetException | IllegalAccessException e) {
                // error handling
            }
        }
    }
}
```

**My Question**: How can I trace `Sample1.logic1()` and `Sample1.logic2()` when they are run through Reflection API?

**Comment by laurit** [comment]
Please provide a minimal sample application that reproduces the issue.

**Reply by rumpfc** [reply]
Here is an initial project for demonstration purposes. I will extend it to use a combination of different tracing tools
[https://github.com/rumpfc/otel-invoke-demo](https://github.com/rumpfc/otel-invoke-demo)

Currently I only set up zipkin, but I will follow up with otel-collector, tempo and grafana. I once detected the issue while pushing traces to otel-collector and tempo. For some reason, I don't see any issues when pushing directly to zipkin.
